### PR TITLE
🐛 Source Stripe: Fix the `Refunds` stream missing data in `Incremental` sync mode

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving:
- https://github.com/airbytehq/airbyte-internal-issues/issues/2450

## How
- Added new event type: `charge.refund.updated` to `Refunds` stream incremental sync mode to retrieve the `updated` refunds
- Changed the `CreatedCursorIncrementalStripeStream` to `IncrementalStripeStream` to be used for the `Refunds` stream and make the `Refunds` stream `concurrent` again
- updated `unit_tests` to reflect the changes

## User Impact
This is the breaking change since the `cursor field` has changed from `created` > `updated.` Resetting of the affected stream is required.

## Can this PR be safely reverted and rolled back?
- [X] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Review by maxi297 — COMMENTED** [review]
Is it a breaking change? I'm asking because the state format is changing from `{"created": 1717598674}` to `{"updated": 1717598674}`

**Review by maxi297 — APPROVED** [review]
Preemptively approving given that [this comment](https://github.com/airbytehq/airbyte/pull/39138#pullrequestreview-2100698146) is addressed

File: airbyte-integrations/connectors/source-stripe/metadata.yaml [hunk 1]
```diff
@@ -36,6 +36,12 @@ data:
       5.0.0:
         message: Version 5.0.0 introduces fixes for the `CheckoutSessions`, `CheckoutSessionsLineItems` and `Refunds` streams. The cursor field is changed for the `CheckoutSessionsLineItems` and `Refunds` streams. This will prevent data loss during incremental syncs. Also, the `Invoices`, `Subscriptions` and `SubscriptionSchedule` stream schemas have been updated.
         upgradeDeadline: "2023-12-11"
+      5.4.0:
+        message: Version 5.4.0 introduces fixes for the `Refunds` streams. The cursor field is changed from `created` to `updated`.
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-stripe/metadata.yaml:40]: I would phrase this as an added feature and not a fix. Like, "`Refunds`, which previously was incremental on the creation date, now tracks updates as well. In order to do that, the cursor field needs to be updates and a full_refresh is recommended (needed?) to get the updates" or something like that

**Review by maxi297 — APPROVED** [review]
Conditional approval on the wording if you think this makes more sense

File: airbyte-integrations/connectors/source-stripe/metadata.yaml (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-stripe/metadata.yaml]: Corrected.

**Pull request merged**